subprojects/matrix-benchmarking-workloads/rhods-notebooks-ux/horreum_lts_store.py

```diff
@@ -20,85 +20,96 @@ def register_lts_metric(cluster_role, metric):
     for (name, query) in metric.items():
         lts_metrics[cluster_role].append((name, query))
 
+
 def _parse_lts_dir(add_to_matrix, dirname, import_settings):
-    with open(f'{dirname}/data.json') as f:
+    with open(dirname / "data.json") as f:
         data = json.load(f)
-        add_to_matrix(_encode_json(data, dirname, data['settings']))
+
+    add_to_matrix(_encode_json(data, dirname, data["settings"]))
 
 
 def _parse_entry(val):
     type_skiplist = [PosixPath, types.FunctionType]
 
     val_type = type(val)
+
     if val_type is datetime.datetime:
         return datetime.datetime.timestamp(val)
+
     elif val_type in [dict, types.SimpleNamespace, common.MatrixEntry, defaultdict, list]:
-        parsed = _decode_ci_items(val)
-        return parsed
+        return _decode_ci_items(val)
+
     elif val_type not in type_skiplist:
         return val
 
 
-def _decode_ci_items(obj):
-    obj_type = type(obj)
-
-    if obj_type == list:
-        return [ _parse_entry(val) for val in obj ]
-    elif obj_type in [types.SimpleNamespace, common.MatrixEntry, dict, defaultdict]:
-        if obj_type in [types.SimpleNamespace, common.MatrixEntry]:
-            obj = vars(obj)
-        res = {}
-        for (key, val) in obj.items():
-            res[key] = _parse_entry(val)
-        return res
-    return obj
-
-
-def _encode_entry(obj):
-    obj_type = type(obj)
-
-    if obj_type == list:
-        return [ _encode_entry(val) for val in obj ]
-    elif obj_type is not dict:
-        return obj
-
-    result = {}    
-    final_type = obj.get("$type", "dict")
-    try:
-        del obj['$type']
-    except Exception:
-        pass
-    
+def _decode_ci_items(src_obj):
+    src_obj_type = type(src_obj)
+
+    if src_obj_type == list:
+        new_obj = [_parse_entry(val) for val in src_obj ]
+
+    elif src_obj_type in [types.SimpleNamespace, common.MatrixEntry, dict, defaultdict]:
+        _src_obj = vars(src_obj)
+            if src_obj_type in [types.SimpleNamespace, common.MatrixEntry] \
+            else src_obj
```

Comment:
I prefer avoiding name shadowing (`obj = vars(obj)` makes the `obj` argument unreachable)